plugins/fluentd_telemetry_plugin/src/streamer.py

```diff
@@ -438,9 +505,20 @@ def _parse_telemetry_csv_metrics_to_json_with_delta(self, data):  # pylint: disa
         is_meta_fields_available = len(self.meta_fields[0]) or len(self.meta_fields[1])
         output = []
 
-        port_id_keys_indices = []
-        for port_id_key in self.port_id_keys:
-            port_id_keys_indices += [i for i, x in enumerate(keys) if x == port_id_key]
+        normal_port_id_keys_indices = []
```

Comment:
Shouldn't you check the xdr_modr here?
You can have different paths and save if checks

Reply:
The XDR headers, such as `port_type` and `sys_image_guid`, only appear when `xdr_mode` is true.
So, checking these headers is enough from my point of view without also checking the xdr_mode